### ISSUE TITLE
Remarks about CodeReference 

### ISSUE THREAD

**Issue by LaurentDardenne** [jhoek/UncommonSense.PowerShell.TypeData#8]
CodeProperty and CodeMethod cmdlet do not use a scriptblock but the type CodeReference.
The declaration of one of these nodes no longer respects the syntax of the 'dsl' :
```Powershell  
Types -PreContent '<?xml version="1.0" encoding="utf-8"?>' {
 _Type ClassesPS.Personne {
   $getter= CodeReference ClassesPS.AdapterCodeProperty DatasGet
   $setter= CodeReference ClassesPS.AdapterCodeProperty DatasSet
   CodeProperty Informations  -GetCodeReference $Getter -SetCodeReference $Setter
 }
}
  
Types {
 _Type System.Xml.XmlNode {
  CodeMethod ToString  (CodeReference Microsoft.PowerShell.ToStringCodemethods XmlNode)
  }
 }
```
For CodeMethod, this would be more 'intuitive' : 
```Powershell  
Types {
 _Type System.Xml.XmlNode {
  CodeMethod ToString  {
     CodeReference Microsoft.PowerShell.ToStringCodemethods XmlNode)
   }
  }
}
```
And may be, mask the use of CodeReference cmdlet : 
```Powershell
Types {
 _Type System.Xml.XmlNode {
   CodeMethod PSAdapted CLRType CLRMethodName
  }
 }
```

**Comment by jhoek** [comment]
Let's first make sure we agree about the basics... :-)

- New-CodeProperty's parameters are already positional. The assumption is that if you specify only one code reference, that will be the getter;
- I feel it makes sense to have separate, positional parameters for the getter and the setter, as opposed to the (theoretical) possibility of having a single script block parameter.
If they use named parameters (as opposed to positional ones), uses can specify the getter and setter in either order. 
- They could even specify only a setter (although I don't see a practical use case), which I feel would be entirely impossible if New-CodeProperty would have a single script block parameter.
- In other words: the script block parameters make sense for situations where a user can specify any number (0..infinity) of subitems. If the numbers of sub-items is known and defined by the domain, I feel normal parameters are the most idiomatic choice.

**Comment by jhoek** [comment]
For New-CodeMethod, allowing the user to provide either a CodeReference instance, *or* a type name + a method name. The two options could be represented by two parameter sets, making them mutually exclusive. 

However, New-CodeProperty also uses CodeReference, and for consistency's sake, would need to do the same, but I feel readability of the calls would suffer.

Please feel free to reopen this issue if you want to discuss it some more, or if there's a serious flaw in my reasoning above! :-)

**Comment by jhoek** [comment]
> They could even specify only a setter (although I don't see a practical use case), which I feel would be entirely impossible if New-CodeProperty would have a single script block parameter.

Please ignore this comment above - a getter is mandatory, a setter is optional. You cannot have a write-only code property.
